design/apps.go

```diff
@@ -0,0 +1,260 @@
+			a.Param("spaceID", d.UUID, "ID of the space")
+		})
+		a.Response(d.OK, simpleSpaceSingle)
+		a.Response(d.NotFound, JSONAPIErrors)
+	})
+
+	a.Action("showSpaceApp", func() {
+		a.Routing(
+			a.GET("/spaces/:spaceID/applications/:appName"),
+		)
+		a.Description("list application")
+		a.Params(func() {
+			a.Param("spaceID", d.UUID, "ID of the space")
+			a.Param("appName", d.String, "Name of the application")
+		})
+		a.Response(d.OK, simpleAppSingle)
+		a.Response(d.NotFound, JSONAPIErrors)
+	})
+
+	a.Action("showSpaceAppDeployment", func() {
+		a.Routing(
+			a.GET("/spaces/:spaceID/applications/:appName/deployments/:deployName"),
+		)
+		a.Description("list pipe element")
+		a.Params(func() {
+			a.Param("spaceID", d.UUID, "ID of the space")
+			a.Param("appName", d.String, "Name of the application")
+			a.Param("deployName", d.String, "Name of the pipe deployment")
+		})
+		a.Response(d.OK, simpleDeploymentSingle)
+		a.Response(d.NotFound, JSONAPIErrors)
+	})
+
+	a.Action("showDeploymentStats", func() {
+		a.Routing(
+			a.GET("/spaces/:spaceID/applications/:appName/deployments/:deployName/stats"),
+		)
+		a.Description("list deployment statistics")
+		a.Params(func() {
+			a.Param("spaceID", d.UUID, "ID of the space")
+			a.Param("appName", d.String, "Name of the application")
+			a.Param("deployName", d.String, "Name of the deployment")
+		})
+		a.Response(d.OK, simpleDeploymentSingle)
+		a.Response(d.NotFound, JSONAPIErrors)
+	})
+
+	a.Action("showDeploymentStatSeries", func() {
+		a.Routing(
+			a.GET("/spaces/:spaceID/applications/:appName/deployments/:deployName/statseries"),
+		)
+		a.Description("list deployment statistics")
+		a.Params(func() {
+			a.Param("spaceID", d.UUID, "ID of the space")
+			a.Param("appName", d.String, "Name of the application")
+			a.Param("deployName", d.String, "Name of the deployment")
+			a.Param("start", d.Integer, "start time in millis")
+			a.Param("end", d.Integer, "end time in millis")
+			a.Param("limit", d.Integer, "maximum number of data points to return")
+		})
+		a.Response(d.OK, simpleDeploymentStatSeries)
+		a.Response(d.NotFound, JSONAPIErrors)
+	})
+
+	a.Action("setDeployment", func() {
+		a.Routing(
+			a.PUT("/spaces/:spaceID/applications/:appName/deployments/:deployName/control"),
+		)
+		a.Description("set deployment pod count")
+		a.Params(func() {
+			a.Param("spaceID", d.UUID, "ID of the space")
+			a.Param("appName", d.String, "Name of the application")
+			a.Param("deployName", d.String, "Name of the deployment")
+			a.Param("podCount", d.Integer, "desired running pod count")
+		})
+		a.Response(d.OK)
+		a.Response(d.NotFound, JSONAPIErrors)
+	})
+
+	a.Action("showSpaceEnvironments", func() {
+		a.Routing(
+			a.GET("/spaces/:spaceID/environments"),
+		)
+		a.Description("list all environments for a space")
+		a.Params(func() {
+			a.Param("spaceID", d.UUID, "ID of the space")
+		})
+		a.Response(d.OK, simpleEnvironmentMultiple)
+		a.Response(d.NotFound, JSONAPIErrors)
```

Comment:
same comment as above about access restriction and `InternalServerError`

Reply:
see above comment